test/with_dummyserver/test_connectionpool.py

```diff
@@ -695,6 +696,16 @@ def test_stream_keepalive(self):
         self.assertEqual(self.pool.num_connections, 1)
         self.assertEqual(self.pool.num_requests, x)
 
+    def test_read_chunked_short_circuit(self):
+        response = self.pool.request(
+            'GET',
+            '/chunked',
+            preload_content=False
+        )
+        response.read()
+        with pytest.raises(StopIteration):
```

Comment:
Wouldn't we want to validate that this _doesn't_ raise? Or would we want more of a test that explicitly closes and then tries to read?

Reply:
I can add a test for closing as well. I lifted this test case from the original issue attached.

Reply:
read_chunked() is a generator so raising StopIteration is the equivalent of an empty string read.

Reply:
Gotcha. Yeah that's my bad; didn't think about it properly.

Reply:
Oh no worries! :) I'll add the additional test case.